packages/protocol/contracts/L1/libs/LibTokenomics.sol

```diff
@@ -180,26 +180,23 @@ library LibTokenomics {
         TaikoData.FeeConfig memory feeConfig,
         uint256 feeBase,
         bool isProposal,
-        uint256 tNow, // seconds
-        uint256 tLast, // seconds
-        uint256 tAvg // milliseconds
-    ) internal pure returns (uint256 newFeeBase, uint256 tRelBp) {
-        if (tAvg == 0 || tNow == tLast) {
+        uint256 timeUsed, // seconds
+        uint256 timeAverage // milliseconds
+    ) internal pure returns (uint256 newFeeBase, uint256 premiumRate) {
+        if (timeAverage == 0) {
             return (feeBase, 0);
         }
-
         unchecked {
-            tAvg = tAvg.min(feeConfig.avgTimeCap);
-            uint256 max = (feeConfig.maxPeriodPctg * tAvg) / 100;
-            uint256 grace = (feeConfig.gracePeriodPctg * tAvg) / 100;
-            uint256 t = ((tNow - tLast) * 1000).max(grace).min(max);
-            tRelBp = (10000 * (t - grace)) / (max - grace); // [0-10000]
-            uint256 alpha = 10000 + (tRelBp * feeConfig.multiplerPctg) / 100;
+            uint p = feeConfig.startBips; // [0-10000]
+            uint a = timeAverage;
+            uint t = (timeUsed * 1000).min(a * 2); // millisconds
+
+            newFeeBase = (feeBase * (10000 + (t * p) / a - p)) / 10000;
 
             if (isProposal) {
-                newFeeBase = (feeBase * 10000) / alpha; // fee
-            } else {
-                newFeeBase = (feeBase * alpha) / 10000; // reward
+                newFeeBase = (feeBase * 2) - newFeeBase;
+            } else if (p > 0) {
+                premiumRate = ((t.max(a) - a) * 10000) / a;
```

Comment:
I have a question to this formula.
Let's say we have 2 scenarios. (In both cases the `startBips` is `2500` = 25% and the `feeBase` is `10`)

1. Quciker than avg (10s )
Average time is 20s (`timeAverage=20.000 ms`) and elapsed time is 10s (`timeUsed = 10.000 ms`). 
Then:
t = (10.000ms ) MIN (20.000 * 2 = 40.000ms) : 10.000MS
newFeeBase = (10 * (10.000 + (10.000 * 2.500 ) / 20.000 - 2.500 ) / 10.000 =  11.42

And in case of proposal it is: `newFeeBase = (10 * 2 ) - 11.42` -> 8.57

2. Slower than avg (50s )
Average time is 20s (`timeAverage=20.000 ms`) and elapsed time is 50s (`timeUsed = 150.000 ms`). 
Then:
t = (50.000ms ) MIN (20.000 * 2 = 40.000ms) : 40.000MS
newFeeBase = (10 * (10.000 + (40.000 * 2.500 ) / 20.000 - 2.500 ) / 10.000 =  15.72

And in case of proposal it is: `newFeeBase = (10 * 2 ) - 15.72` -> 4.28

If the calcs are correct it means the initial baseFee can only decrease or stagnate.
Are my calculations correct ?

Reply:
- scenario 1 (at time 10sec):
    - fee for proposer is 11.25, 
    - reward for prover is 8.65.
- scenario 2 (at time 50sec):
    - fee for proposer is 7.5
    - reward for prover is 12.5.